### PR TITLE
Added OpsGenie notifier

### PR DESCRIPTION
**Pull request by vladshub**

### CODE REVIEW

**Comment by ericwestfall** [comment]
How exciting, our first new notifier! Welcome to the project @vladshub and thank you so much for contributing! 

@jrasell is going to review this for you and get you some feedback as soon as he can.

**Comment by jrasell** [comment]
Thanks @vladshub for this. I have a couple of comments which I will make here then some additional ones that I will relate to code within the PR.

Could you please add a description to the PR and also add any testing outputs and examples you have during any testing? This will help us quickly and accurately update our documentation with examples without the immediate need to open an opsgenie account.  @ericwestfall and I realised our contributing guide was a little lacking, so have updated this in a hope to make the PR process easier in the future.

Thanks again!!

File: command/base/config_parse_test.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func TestConfigParse_LoadConfigFile(t *testing.T) {
 
     notification {
       pagerduty_service_key = "thistooisafakekey"
+	  opsgenie_service_key  = "thisisafakeapikey"
```
Comment (jrasell) [inline command/base/config_parse_test.go]: Pedantic I know; but can this be indented correctly?

File: main.go [hunk 2]
```diff
@@ -40,13 +40,13 @@ func RunCustom(args []string, commands map[string]cli.CommandFactory) int {
 		}
 	}
 
-	cli := &cli.CLI{
+	replicatorCli := &cli.CLI{
```
Comment (jrasell) [inline main.go]: Is there any reason why this needed to change as it shouldn't be related to adding a notifier? If not, could this please be changed back.

**Comment by vladshub** [comment]
@jrasell thanks for the review changes are committed.
Reguarding the change in main.go the variable named cli collides with the package cli...

File: command/agent/agent.go [hunk 3]
```diff
@@ -228,6 +229,19 @@ func (c *Command) setupNotifier(config *structs.Notification) (err error) {
 		config.Notifiers = append(config.Notifiers, pd)
 	}
 
+	// Configure the OpsGenie notifier.
+	if config.OpsGenieAPIKey != "" {
+
+		ogConfig := make(map[string]string)
```
Comment (jippi) [inline command/agent/agent.go]: why not reduce this to `notifier.NewProvider("opsgenie",  make(map[string]string){"OpsGenieAPIKey": config.OpsGenieAPIKey})` ?

Comment (vladshub) [reply command/agent/agent.go]: @jippi code readability and it is the same setup done in the pager duty section ...

**Pull request merged**